static/app/components/profiling/threadSelector.tsx

```diff
@@ -1,226 +1,87 @@
-import * as React from 'react';
-import {css} from '@emotion/react';
-import styled from '@emotion/styled';
-import Fuse from 'fuse.js';
+import {useCallback, useMemo} from 'react';
 
-import AutoComplete from 'sentry/components/autoComplete';
-import Button from 'sentry/components/button';
-import Input from 'sentry/components/forms/controls/input';
-import {IconCheckmark} from 'sentry/icons';
-import {t} from 'sentry/locale';
-import space from 'sentry/styles/space';
+import CompactSelect from 'sentry/components/forms/compactSelect';
+import {IconList} from 'sentry/icons';
+import {SelectValue} from 'sentry/types';
+import {defined} from 'sentry/utils';
 import {ProfileGroup} from 'sentry/utils/profiling/profile/importProfile';
 import {Profile} from 'sentry/utils/profiling/profile/profile';
-import useOnClickOutside from 'sentry/utils/useOnClickOutside';
-
-const sortProfiles = (profiles: ReadonlyArray<Profile>): ProfileGroup['profiles'] => {
-  return [...profiles].sort((a, b) => {
-    if (!b.duration) {
-      return -1;
-    }
-    if (!a.duration) {
-      return 1;
-    }
-
-    if (a.name.startsWith('(tid') && b.name.startsWith('(tid')) {
-      return -1;
-    }
-    if (a.name.startsWith('(tid')) {
-      return -1;
-    }
-    if (b.name.startsWith('(tid')) {
-      return -1;
-    }
-    if (a.name.includes('main')) {
-      return -1;
-    }
-    if (b.name.includes('main')) {
-      return 1;
-    }
-    return a.name > b.name ? -1 : 1;
-  });
-};
 
 interface ThreadSelectorProps {
   activeProfileIndex: ProfileGroup['activeProfileIndex'];
   onProfileIndexChange: (index: number) => void;
   profileGroup: ProfileGroup;
 }
 
-function ThreadMenuSelector(props: ThreadSelectorProps): React.ReactElement | null {
-  const containerRef = React.useRef<HTMLDivElement>(null);
-  const [open, setOpen] = React.useState<boolean>(false);
-
-  const handleSelectItem = React.useCallback(
-    (p: Fuse.FuseResult<Profile & {index: number}>) => {
-      const index = props.profileGroup.profiles.findIndex(
-        profile => profile.name === p.item.name
-      );
-      if (index === -1) {
-        return;
+function ThreadMenuSelector({
+  activeProfileIndex,
+  onProfileIndexChange,
+  profileGroup,
+}: ThreadSelectorProps) {
+  const options: SelectValue<number>[] = useMemo(() => {
+    return profileGroup.profiles
+      .map((profile, i) => ({
+        name: profile.name,
+        duration: profile.duration,
+        index: i,
+      }))
+      .sort(compareProfiles)
+      .map(item => ({label: item.name, value: item.index}));
+  }, [profileGroup]);
+
+  const handleChange = useCallback(
+    opt => {
```

Comment:
Nit, but the opt here is any type. Can you add the type from the select component or something? Maybe just type handleChange: CompactSelectProps[onChange] or something